Gen weights

HerrHorizontal opened the pull request:
Added feature to add genWeights (PS, scales, PDF) into the lheWeight list.

It has been tested and proven to work. See:
```bash
/portal/ekpbms1/home/mhorzela/CMSSW_10_6_0_skimUL2017/src/Kappa/Skimming/zjet/configs/jec17ul/1060_DYJetsToLL_Summer19-madgraphMLM_mc2017_realistic_v6-v3/test/106X_mc2017_realistic_v6
```

Commit:
TOBETESTED: Including ISR/FSR weights to lheWeight vector

Commit:
test prefixes

Commit:
corrected push weightname

Commit:
Include ISR/FSR and other PS weights

Commit:
Corrected logic for obtaining lheWeight from KGenEventInfo

mschnepf commented on DataFormats/interface/KInfo.h:
```diff
@@ -174,7 +174,7 @@ struct KGenEventInfo : public KEventInfo
 	{
 		if(failOnError)
-			assert(lheWeight[index] < float(-998.9) && lheWeight[index] > float(999.1) ); // the user tried to access something that has not been properly filled during the skim
+			assert(!(lheWeight[index] < float(-998.9) || lheWeight[index] > float(999.1)) ); // the user tried to access something that has not been properly filled during the skim
```
Is the cut value abs(999)? Than you could use <=-999 and >=999 instead of these floats

HerrHorizontal replied:
This might actually stongly depend on the MC sample. So my suggestion would be to drop it, or change the default behaviour to `failOnError = false`. There are only a few cases where it is resonable to cut on the eventWeight.

HerrHorizontal replied:
I change the default behaviour

mschnepf commented on Producers/interface/KGenInfoProducer.h:
```diff
@@ -77,7 +86,7 @@ class KGenInfoProducer : public KInfoProducer<Tmeta>
 
 	virtual bool onFirstEvent(const edm::Event &event, const edm::EventSetup &setup)
-	{
+	{		
```
```suggestion
	{
```

mschnepf reviewed (commented):
Some small comments. Looks good

HerrHorizontal committed:
Cosmetic update of Producers/interface/KGenInfoProducer.h

Co-authored-by: Matthias Schnepf <maschnepf@schnepf-net.de>

HerrHorizontal committed:
changed work dir to /work

HerrHorizontal committed:
Changed default behaviour of getLheWeight

It might lead to unphysical observables to cut the GEN weights. Cutting the weight is fine only if the user knows exactly what he does.
